### PR TITLE
refactor(world): change most values from ints to doubles

### PR DESCRIPTION
**Pull request by anarchuser**
* Change all positions, energy values and vision ranges from `int` to `double`
* Change tick counter and IDs from `int` to `long`

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #include "main.h"
 
-int main(int argc, char * argv[]) {
+int main() {
     Controller controller = Controller(WIDTH, HEIGHT, ENERGY);
     controller.init();
     controller.run();
-    return 0;
```
Comment (AtomToast) [inline src/main.cpp:8]: Why are you removing the return statement?

**Comment by AtomToast** [comment]
Also, does c++ assume the long to be a long int if you don't specify anything? Since it could also be a long double

**Comment by anarchuser** [comment]
> Also, does c++ assume the long to be a long int if you don't specify anything? Since it could also be a long double

yes.
`long` == `long int` == `signed long` == `singed long int`
long doubles are rarely used; it wouldn't make sense to allow this ambiguity.
you're right that long is actually just a modifier, not a type, but for simplicity it exists as its own data type

see [cppreference.com](https://en.cppreference.com/w/cpp/language/types)

**Comment by xeruf** [comment]
See https://github.com/Xerus2000/eve/pull/21

**Comment by anarchuser** [comment]
this, too, doesn't depend on another PR

**Review by AtomToast — APPROVED** [review]
I don't know what the hell is going on

**Pull request merged**